packages/react-dom/src/client/ReactDOMFiberInput.js

```diff
@@ -205,16 +205,28 @@ export function updateWrapper(element: Element, props: Object) {
   }
 }
 
-export function postMountWrapper(element: Element, props: Object) {
+export function postMountWrapper(
+  element: Element,
+  props: Object,
+  isHydrating: boolean,
+) {
   const node = ((element: any): InputWithWrapperState);
 
   if (props.hasOwnProperty('value') || props.hasOwnProperty('defaultValue')) {
     const initialValue = '' + node._wrapperState.initialValue;
-    const currentValue = node.value;
+    let currentValue;
+    if (isHydrating) {
+      currentValue = node.value;
+    } else {
+      // With range inputs node.value may be a default value calculated from the
+      // min/max attributes. This ensures that node.value is set with the correct
+      // value coming from props.
+      currentValue = props.type === 'range' ? '' : node.value;
+    }
 
     // Do not assign value if it is already set. This prevents user text input
     // from being lost during SSR hydration.
-    if (currentValue === '') {
+    if (!node.hasAttribute('value')) {
       // Do not re-assign the value property if there is no change. This
       // potentially avoids a DOM write and prevents Firefox (~60.0.1) from
       // prematurely marking required inputs as invalid
```

Comment:
Woah nice. You added the `isHydrating` part! Very cool. I have a few questions:

1. Can you use `isHydrating` as the condition on line 129, like `if (!isHydrating)?`
2. Are the lines between (218-225) necessary? I think the bug was that `range` inputs never reported `""`, as required by the old code on line 229, that prevented the value assignment from happening. 

Basically, I think the only change we need to make is on line 229:

```
if (!isHydrating) {
  // ....
}
```

Reply:
Oh, you're absolutely right! I'll update it right away.